### PR TITLE
Implememt API and commad for enum process handles

### PR DESCRIPTION
**Pull request by dennisieur**
1. Adding API for enum process handles
    * enum process handles
    * enum process handles with options
2. Adding commands for listing process handles
    * `list_handles` can parse with options
        * `list_handles [ -c CR3 | -e eprocess | -p pid | -t type | -f full_detail | -d detail ]`
    * `list_handles_types` can list all handles types
3. Revise commands `list_process` ( function "memfrs_enum_proc_list" in ext/memfrs/memfrs.c )
    * `list_process` divide to command part and API part
4. Remove task case for "memfrs_enum_proc_list" in ext/memfrs/memfrs.c

### CODE REVIEW

**Comment by GlacierW** [comment]
The QEMU hangs by repeatedly input the memfrs command `list_process`.

File: ext/memfrs/memfrs.h [hunk 1]
```diff
@@ -68,11 +99,61 @@ extern json_object* memfrs_q_globalvar(const char* gvar_name);
...
+extern UT_array* enum_proc_handles(uint64_t target_cr3, uint64_t kpcr_ptr, CPUState *cpu);
+
+Eumerate the running process' handles
+
+INPUT:     uint64_t target_cr3       the target cr3. if target_cr3=0x0, stored handles of all process
+           uint64_t kpcr_ptr,        the address of _KPCR struct
+           CPUState *cpu,            the pointer to current cpu
+OUTPUT:    UT_array*                 return a handles structire of UTarray
+*******************************************************************/
+extern UT_array* enum_proc_handles(uint64_t target_cr3, uint64_t kpcr_ptr, CPUState *cpu);
```
Comment (dennisieur) [inline ext/memfrs/memfrs.h]: Fixed it.

**Comment by dennisieur** [comment]
Fixed these issue
1. The QEMU hangs by repeatedly input the memfrs command list_process. 
2. Please revise to follow the API naming convention (prefix with extension name)

**Comment by GlacierW** [comment]
segfault on `list_handles`

**Pull request closed without merging**